app/fileManager.js

```diff
@@ -59,17 +69,61 @@ export default function(state, emitter) {
   function render() {
     emitter.emit('render');
   }
+  /*
+
+  function updateDloads() {
+    emitter.emit('updateDloads');
+  }
+*/
+
+  const dlimitArray = [];
+  const fileIdArray = [];
+  function ifRerender(file) {
+    if (fileIdArray.indexOf(file.id) === -1) {
+      const fileDetails = {
+        fileId: file.id,
+        dlimit: file.dlimit,
+        dtotal: file.dtotal
+      };
+      dlimitArray.push(fileDetails);
+      return false;
+    } else {
+      for (const fileDetail of dlimitArray) {
+        if (file.id === fileDetail.fileId) {
+          if (
+            file.dlimit !== fileDetail.dlimit ||
+            file.dtotal !== fileDetail.dtotal
+          ) {
+            fileDetail.dtotal = file.dtotal;
+            fileDetail.dlimit = file.dlimit;
+            return true;
+          } else {
+            return false;
+          }
+        }
+      }
+    }
+  }
```

Comment:
This entire functions controls the possibilities of `rerender` which include whether a dlimit or dcount change has occurred.

All existing files are stored in `fileIdArray`.

Details of all `dcount` and `dlimit` of a given file is stored in `dlimitArray`. And then if it has changed and then rerender is called. Pretty easy to understand though.

Reply:
I'd prefer not to have `dlimitArray` and `fileIdArray` at the scope where they are. It keeps them alive longer than necessary. I think there's a way you can compare the old/new values without them. Maybe while you iterating over the files in `checkFiles`?

Reply:
Yeah that's true, I've implemented it there only now.